### ISSUE TITLE
[wgsl-in] Missing parentheses around dereference

### ISSUE THREAD

**Issue by Herschel** [gfx-rs/naga#1356]
https://github.com/gfx-rs/naga/pull/1348 fixed this for wgsl-out, but the following still incorrectly validates:

```
[[stage(vertex)]]
fn main() -> [[builtin(position)]] vec4<f32> {
    var v: vec4<f32> = vec4<f32>(1.0, 1.0, 1.0, 1.0);
    let pv = &v;
    let a = *pv[3]; // Problematic line
    return v;
}
```

```
cargo run -- test.wgsl
```

[According to the WGSL grammar](https://www.w3.org/TR/WGSL/#expression-grammar), [] should bind more tightly than *, so `*pv[3]` should not pass validation. The correct code would be `(*pv)[3]`.

naga @ https://github.com/gfx-rs/naga/commit/0e66930aff1340d5c636df616d573cc326702158

**Comment by jimblandy** [comment]
I've reproduced this. Indeed, `*pv[3]` parenthesizes as `*(pv[3])`, which is ill-typed.

Naga's WGSL front end doesn't follow the grammar in the spec very closely, so I think there are a number of problems along these lines. See also #1352.

**Comment by jimblandy** [comment]
This was not fixed by #1360. The [approach](https://github.com/gfx-rs/naga/blob/933ecc4a443cb882bce0283fe77897bba6007080/src/front/wgsl/mod.rs#L2083-L2087) I took to handling `let` bindings is incorrect.
